### PR TITLE
fix: Add check for go modules

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add a check in the CI to ensure that go modules are up to date.

This will catch issues like #718 from recurring.

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Failing on the CI here: https://github.com/Mellanox/network-operator/actions/runs/7275136310/job/19822555302?pr=719

Should pass on current version.

**Review by ykulazhenkov — APPROVED** [review]

**Review by rollandf — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -46,6 +46,16 @@ jobs:
         uses: actions/checkout@v3
       - name: check manifests
         run: make check-manifests
+  diff-modules:
```
Comment (e0ne) [inline .github/workflows/build.yml:49]: We need to switch to Ubuntu 22.04 for all workflows. It should be done in a separate PR

**Review by e0ne — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]
